ogc/src/main/scala/geotrellis/server/ogc/wms/WmsModel.scala

```diff
@@ -16,62 +16,78 @@
+  GeoTrellisOgcSource,
+  MapAlgebraSource,
+  OgcSource,
+  OgcSourceRepository,
+  SimpleSource
+}
 import geotrellis.server.ogc.style._
 import geotrellis.server.ogc.wms.WmsParams.GetMap
 import jp.ne.opt.chronoscala.Imports._
 
-
 /** This class holds all the information necessary to construct a response to a WMS request */
 case class WmsModel(
-  serviceMeta: opengis.wms.Service,
-  parentLayerMeta: WmsParentLayerMeta,
-  sources: OgcSourceRepository
+    serviceMeta: opengis.wms.Service,
+    parentLayerMeta: WmsParentLayerMeta,
+    sources: OgcSourceRepository
 ) {
 
-  private lazy val times: List[ZonedDateTime] = sources.store.flatMap(_.time)
-
-  lazy val temporalRange: Option[OgcTimeInterval] = {
-    times match {
-      case Nil => None
-      case head :: Nil => OgcTimeInterval(head).some
-      case list => OgcTimeInterval(list.min, Some(list.max), None).some
+  def timeInterval: Option[OgcTimeInterval] = {
+    sources.store.flatMap(_.timeInterval).reduceOption { (a, b) =>
+      val times = List(Some(a.start), a.end, Some(b.start), b.end).flatten
+      OgcTimeInterval(times.min, Some(times.max), None)
     }
   }
 
   /** Take a specific request for a map and combine it with the relevant [[OgcSource]]
-   *  to produce an [[OgcLayer]]
-   */
-  def getLayer(params: GetMap): List[OgcLayer] = {
-    val p = temporalRange match {
-      // Add default time to GetMap params if this is a temporal layer and the user didn't request a specific time
-      case Some(timeInterval) if params.time.isEmpty =>
-        params.copy(time = OgcTimeInterval(timeInterval.start).some)
-      case _ => params
-    }
-    for {
-      supportedCrs <- parentLayerMeta.supportedProjections.find(_ == p.crs).toList
-      source       <- sources.find(p.toQuery)
-    } yield {
-      val styleName: Option[String] =
-        p.styles
-          .headOption
-          .filterNot(_.isEmpty)
-          .orElse(source.defaultStyle)
-      val style: Option[OgcStyle] =
-        styleName.flatMap { name => source.styles.find(_.name == name) }
-      source match {
-        case MapAlgebraSource(name, title, rasterSources, algebra, defaultStyle, styles) =>
-          val simpleLayers = rasterSources.mapValues { rs =>
-            SimpleOgcLayer(name, title, supportedCrs, rs, style)
+    *  to produce an [[OgcLayer]]
+    */
+  def getLayer(p: GetMap): List[OgcLayer] = {
+    parentLayerMeta.supportedProjections
+      .find(_ == p.crs)
+      .fold[List[OgcLayer]](List()) { supportedCrs =>
+        sources.find(p.toQuery).map { source =>
+          val styleName: Option[String] =
+            p.styles.headOption
+              .filterNot(_.isEmpty)
+              .orElse(source.defaultStyle)
+          val style: Option[OgcStyle] =
+            styleName.flatMap { name =>
+              source.styles.find(_.name == name)
+            }
+          source match {
+            case MapAlgebraSource(name,
+                                    title,
```

Comment:
How do you feel about such formating? Looks a bit unreadble to me ._.

Also it looks like the old `for` syntax was okay; if you don't like it how about smth like:
![image](https://user-images.githubusercontent.com/4929546/77451102-99d64800-6dca-11ea-8822-d83e282e68dc.png)

Reply:
Oof. Yeah the formatting here sucks. I hate that the intellij scalafmt formatter defaults to indenting parameters like that.

Updated